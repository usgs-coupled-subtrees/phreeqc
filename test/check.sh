#!/bin/sh
#
#  check.sh:  differential file comparator (diff) run to compare original
#             output files against newly created output files
#
# output file names must follow format "nameX.sufx" where:  "name" is
# "test" by default or can be supplied as argument(s) to check.sh; all
# numbers X must appear in "for Test" loop list below; all suffixes must 
# appear in "for Sufx" loop list below
#
#  04/28/92, mygoze, initial coding
#  06/10/92, mygoze, added in NameList, Name, Sufx variables and "for Name",
#                    "for Sufx" loops to make check.sh more versatile 
#  02/10/00, mcrouse, last modified

DIVD=----------------------------------------
TOPDIR=..
DATA=$TOPDIR/examples

# set list of names of files to be tested; name will be "test" by default
NameList=${@:-test}

# delete old output file  
if [ -f check.out ]; then rm check.out; fi

for Namevar in $NameList
do
  for Test in 1 2 3 4 5 6 6A-B 6C 7 8 9 10 11 11adv 11trn 12 12a \
	      13a 13b 13c 14 15 16 17 18
  do
    for Sufx in out sel
    do
      if [ -f $DATA/$Namevar$Test.$Sufx -a -f $Namevar$Test.$Sufx ]
      then
      # do comparison only if both orig. and new output files exist
        echo $DIVD$DIVD | tee -a check.out
        echo "comparison of $DATA/$Namevar$Test.$Sufx with $Namevar$Test.$Sufx" \
             | tee -a check.out

        if diff -w $DATA/$Namevar$Test.$Sufx $Namevar$Test.$Sufx >> check.out
        then
          echo FILES ARE IDENTICAL | tee -a check.out
        else
          echo FILES DIFFER:  see file check.out for differences
        fi
      fi
    done
  done
done
