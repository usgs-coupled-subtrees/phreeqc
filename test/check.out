--------------------------------------------------------------------------------
comparison of ../examples/ex1.out with ex1.out
1c1
<    Input file: ./ex1
---
>    Input file: ../examples/ex1
--------------------------------------------------------------------------------
comparison of ../examples/ex2.out with ex2.out
1c1
<    Input file: ./ex2
---
>    Input file: ../examples/ex2
--------------------------------------------------------------------------------
comparison of ../examples/ex2.sel with ex2.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex3.out with ex3.out
1c1
<    Input file: ./ex3
---
>    Input file: ../examples/ex3
--------------------------------------------------------------------------------
comparison of ../examples/ex4.out with ex4.out
1c1
<    Input file: ./ex4
---
>    Input file: ../examples/ex4
--------------------------------------------------------------------------------
comparison of ../examples/ex5.out with ex5.out
1c1
<    Input file: ./ex5
---
>    Input file: ../examples/ex5
--------------------------------------------------------------------------------
comparison of ../examples/ex5.sel with ex5.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex6.out with ex6.out
1c1
<    Input file: ./ex6
---
>    Input file: ../examples/ex6
--------------------------------------------------------------------------------
comparison of ../examples/ex6A-B.sel with ex6A-B.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex6C.sel with ex6C.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex7.out with ex7.out
1c1
<    Input file: ./ex7
---
>    Input file: ../examples/ex7
--------------------------------------------------------------------------------
comparison of ../examples/ex7.sel with ex7.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex8.out with ex8.out
1c1
<    Input file: ./ex8
---
>    Input file: ../examples/ex8
--------------------------------------------------------------------------------
comparison of ../examples/ex8.sel with ex8.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex9.out with ex9.out
1c1
<    Input file: ./ex9
---
>    Input file: ../examples/ex9
--------------------------------------------------------------------------------
comparison of ../examples/ex9.sel with ex9.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex10.out with ex10.out
1c1
<    Input file: ./ex10
---
>    Input file: ../examples/ex10
--------------------------------------------------------------------------------
comparison of ../examples/ex10.sel with ex10.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex11.out with ex11.out
1c1
<    Input file: ./ex11
---
>    Input file: ../examples/ex11
--------------------------------------------------------------------------------
comparison of ../examples/ex11adv.sel with ex11adv.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex11trn.sel with ex11trn.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex12.out with ex12.out
1c1
<    Input file: ./ex12
---
>    Input file: ../examples/ex12
--------------------------------------------------------------------------------
comparison of ../examples/ex12.sel with ex12.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex12a.out with ex12a.out
1c1
<    Input file: ./ex12a
---
>    Input file: ../examples/ex12a
--------------------------------------------------------------------------------
comparison of ../examples/ex12a.sel with ex12a.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex13a.out with ex13a.out
1c1
<    Input file: ./ex13a
---
>    Input file: ../examples/ex13a
--------------------------------------------------------------------------------
comparison of ../examples/ex13a.sel with ex13a.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex13b.out with ex13b.out
1c1
<    Input file: ./ex13b
---
>    Input file: ../examples/ex13b
--------------------------------------------------------------------------------
comparison of ../examples/ex13b.sel with ex13b.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex13c.out with ex13c.out
1c1
<    Input file: ./ex13c
---
>    Input file: ../examples/ex13c
--------------------------------------------------------------------------------
comparison of ../examples/ex13c.sel with ex13c.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex14.out with ex14.out
1c1
<    Input file: ./ex14
---
>    Input file: ../examples/ex14
--------------------------------------------------------------------------------
comparison of ../examples/ex14.sel with ex14.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex15.out with ex15.out
1c1
<    Input file: ./ex15
---
>    Input file: ../examples/ex15
3c3
< Database file: ./ex15.dat
---
> Database file: ../examples/ex15.dat
--------------------------------------------------------------------------------
comparison of ../examples/ex15.sel with ex15.sel
FILES ARE IDENTICAL
--------------------------------------------------------------------------------
comparison of ../examples/ex16.out with ex16.out
1c1
<    Input file: ./ex16
---
>    Input file: ../examples/ex16
--------------------------------------------------------------------------------
comparison of ../examples/ex17.out with ex17.out
1c1
<    Input file: ./ex17
---
>    Input file: ../examples/ex17
--------------------------------------------------------------------------------
comparison of ../examples/ex18.out with ex18.out
1c1
<    Input file: ./ex18
---
>    Input file: ../examples/ex18
